src/main/java/com/openwt/blog/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.openwt.blog.service.impl;
+
+import com.openwt.blog.exception.NotFoundException;
+import com.openwt.blog.model.dto.UserDTO;
+import com.openwt.blog.model.user.User;
+import com.openwt.blog.repository.RoleRepository;
+import com.openwt.blog.repository.UserRepository;
+import com.openwt.blog.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private RoleRepository roleRepository;
+
+    @Override
+    public User register(UserDTO userDTO) {
+        return addUser(userDTO, "ROLE_USER");
+    }
+
+    @Override
+    public List<User> get() {
+        return userRepository.findAll();
+    }
+
+    @Override
+    public Optional<User> get(long id) {
```

Comment:
Hàm này thực tế luôn trả về Object user (vì nếu user không tồn tại thì đã throw exception) nên phải khai báo nó là 
```suggestion
    public User get(long id) {
```

Reply:
Done